packages/middleware-location-constraint/src/index.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  InitializeHandler,
+  InitializeMiddleware,
+  InitializeHandlerArguments,
+  InitializeHandlerOptions,
+  InitializeHandlerOutput,
+  MetadataBearer,
+  Pluggable
+} from "@aws-sdk/types";
+
+export function locationConstraintMiddleware(): InitializeMiddleware<any, any> {
+  return <Output extends MetadataBearer>(
+    next: InitializeHandler<any, Output>
+  ): InitializeHandler<any, Output> => async (
+    args: InitializeHandlerArguments<any>
+  ): Promise<InitializeHandlerOutput<Output>> => {
+    const { region, CreateBucketConfiguration } = args.input;
```

Comment:
Is region an actual user input parameters? Shouldn't this be in the client config?

Reply:
Yes. Fixed in 342e6d65812011cb6a0b4d743b84bf9bf40f2b77